Initial docs for the csp adapter

MbolotSuse opened the pull request:
Starting docs for the csp-adapter.

Closes #4131

MbolotSuse commented:
I've marked this as a WIP for now. While I think that the overall content is fine, it could probably use some screenshots here and there to better identify a couple of steps (I'm primarily thinking that the "Locate the product in the Marketplace" and "Gain access to the local cluster" steps should have pictures).

MbolotSuse commented:
Waiting on rancher/dashboard#4916 so I can get screenshots for how to download the supportconfig bundle. After that, will remove the WIP and move toward getting this merged in.

MbolotSuse commented:
Removing the WIP - these docs are now ready for review @jtravee @btat please let me know if you have questions/concerns on these changes. Would like them to get moved to staging in the near future so that other business partners (product) can take a look.

jtravee commented on content/rancher/v2.6/en/cloud-marketplace/supportconfig/_index.md:
```diff
@@ -0,0 +1,54 @@
+
+![Get Support]({{<baseurl>}}/img/rancher/generate-support-config.png)
+
+## Accessing without rancher
```
```suggestion
### Accessing without Rancher
```

jtravee replied:
Keeping the header sizes consistent.

MbolotSuse replied:
Thank you for the suggestion, change made

jtravee commented on content/rancher/v2.6/en/cloud-marketplace/supportconfig/_index.md:
```diff
@@ -0,0 +1,54 @@
+
+> **Note:** Only admin users can generate/download supportconfig bundles, regardless of method 
+
+## Accessing through Rancher
```
```suggestion
### Accessing through Rancher
```

MbolotSuse replied:
Thank you for the suggestion, change made

jtravee commented on content/rancher/v2.6/en/cloud-marketplace/supportconfig/_index.md:
```diff
@@ -0,0 +1,54 @@
+
+## Accessing without rancher
+
+First, generate a kubeconfig for the cluster rancher is installed on. 
```
```suggestion
First, generate a kubeconfig for the cluster that Rancher is installed on. 
```

jtravee commented on content/rancher/v2.6/en/cloud-marketplace/supportconfig/_index.md:
```diff
@@ -0,0 +1,54 @@
+
+First, generate a kubeconfig for the cluster rancher is installed on. 
+
+> **Note:** If rancher is down, you will not be able to use a kubeconfig token generated by rancher to access the cluster.
```
```suggestion
> **Note:** If Rancher is down, you will not be able to use a kubeconfig token generated by Rancher to access the cluster.
```

MbolotSuse replied:
Thank you for the suggestion, change made

jtravee commented on content/rancher/v2.6/en/cloud-marketplace/supportconfig/_index.md:
````diff
@@ -0,0 +1,54 @@
+mkdir temp && cd temp
+```
+
+Then, retrieve the supportconfig bundle.
````
```suggestion
Then, retrieve the supportconfig bundle:
```

MbolotSuse replied:
Thank you for the suggestion, change made

jtravee commented on content/rancher/v2.6/en/cloud-marketplace/aws/_index.md:
```diff
@@ -0,0 +1,32 @@
+- You must be running rancher v2.6.6 or higher
+- Rancher must be deployed with additional metrics enabled
+- Rancher must be installed on an EKS cluster
+- You must purchase at least one entitlement to rancher support through AWS Marketplace.
```
Adding periods for consistency.

MbolotSuse replied:
Thank you for the suggestion, change made.

btat commented on content/rancher/v2.6/en/cloud-marketplace/_index.md:
```diff
@@ -0,0 +1,8 @@
+
+Rancher offers integration with cloud marketplaces to easily purchase support for installations hosted on certain cloud providers. In addition, this integration also provides the ability to generate a supportconfig bundle which can be provided to rancher support. 
+
+At this time, only AWS is supported for this integration.
```
```suggestion
As of Rancher v2.6.7, only AWS is supported for this integration.
```

With the way we currently do versioning on our docs `at this time` is a bit ambiguous.

MbolotSuse replied:
Thank you for the suggestion, change made.

btat commented on content/rancher/v2.6/en/cloud-marketplace/aws/_index.md:
```diff
@@ -0,0 +1,32 @@
+
+# Limitations
+
+- You must be running rancher v2.6.6 or higher
```
```suggestion
- You must be running Rancher v2.6.7 or higher.
```

Should this be 2.6.7 since the current 2.6.6 is being renamed?

MbolotSuse replied:
Yep - these were written before that scheduling change was made. I've made this change and will do a look around for other places that may also have this error. Good catch, thank you.

btat commented on content/rancher/v2.6/en/cloud-marketplace/aws/_index.md:
```diff
@@ -0,0 +1,32 @@
+
+**Can I purchase support for more nodes later on?**
+
+Yes. Simply go to the AWS marketplace entry that you used to initially purchase support and increase the number of entitlements.
```
```suggestion
Yes. Simply go to the AWS Marketplace entry that you used to initially purchase support and increase the number of entitlements.
```

MbolotSuse replied:
Thank you for the suggestion, change made.

btat commented on content/rancher/v2.6/en/cloud-marketplace/aws/prerequisites/_index.md:
```diff
@@ -0,0 +1,97 @@
+> **Note:** Each entitlement grants access to support for a certain amount of nodes. You can purchase more licenses as necessary later on.
+
+### 2. Create an EKS Cluster
+Follow the [rancher docs]({{<baseurl>}}/rancher/v2.6/en/installation/install-rancher-on-k8s/amazon-eks/) to create an EKS cluster. When you get to the final step to install rancher, stop and return to this page. This cluster will need to meet the following requirements:
```
`Follow the [rancher docs]` -> `Follow the [Rancher docs]`

MbolotSuse replied:
Thank you for the suggestions, change made

btat commented on content/rancher/v2.6/en/cloud-marketplace/aws/prerequisites/_index.md:
```diff
@@ -0,0 +1,97 @@
+
+### 3. Install Rancher
+
+In addition to the options specified to install rancher in step 8 of the [rancher docs]({{<baseurl>}}/rancher/v2.6/en/installation/install-rancher-on-k8s/amazon-eks/), you will also need to enable extra metrics. 
```
`in the [rancher docs]` -> `in the [Rancher docs]`

MbolotSuse replied:
Thank you for the suggestion, change made

btat commented on content/rancher/v2.6/en/cloud-marketplace/aws/prerequisites/_index.md:
````diff
@@ -0,0 +1,97 @@
+    value: "true"
+```
+
+You will also need to install rancher version 2.6.6 or higher.
````
```suggestion
You will also need to install Rancher version v2.6.7 or higher.
```

Should this be 2.6.7 since the current 2.6.6 is being renamed?

MbolotSuse replied:
Thank you for the suggestions, changes made

btat commented on content/rancher/v2.6/en/cloud-marketplace/supportconfig/_index.md:
```diff
@@ -0,0 +1,54 @@
+
+This will create a `supportconfig_rancher.tar` file in your current directory.
+
+Users who run these commands on Mac may experience issues due incompatibilities between gnu-tar and bsd-tar. If support has issues reading a supportconfig that you produce, try using the below command after making gnu-tar accessible as `gtar` on your path:
```
```suggestion
Users who run these commands on Mac may experience issues due to incompatibilities between gnu-tar and bsd-tar. If support has issues reading a supportconfig that you produce, try using the below command after making gnu-tar accessible as `gtar` on your path:
```

MbolotSuse replied:
Thank you for the suggestion, change made

btat commented on content/rancher/v2.6/en/cloud-marketplace/supportconfig/_index.md:
```diff
@@ -0,0 +1,54 @@
+
+In the next page, click on the `Generate Support Config` button.
+
+> **Note:** If the adapter is not installed, the option to generate the support-config bundle will not be present. You must install the csp adapter to generate a support-config bundle.
```
```suggestion
> **Note:** If the adapter is not installed, the option to generate the supportconfig bundle will not be present. You must install the CSP adapter to generate a supportconfig bundle.
```

MbolotSuse replied:
Thank you for the suggestion, change made

btat commented on content/rancher/v2.6/en/cloud-marketplace/aws/common-issues/_index.md:
```diff
@@ -0,0 +1,29 @@
+weight: 4
+---
+
+**After installing the adapter, a banner message appeared in rancher which says "AWS Marketplace Adapter: Unable to run the adapter, please check the adapter logs"**
```
`appeared` -> `appears` to match tense with the other entry on the page.

MbolotSuse replied:
Thank you for the suggestion, change made

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/adapter-install/_index.md:
```diff
@@ -0,0 +1,95 @@
+
+Next, install the CSP adapter. You must specify several values, including: the account number, and the name of the role created in the pre-requisites.
+
+> **Note:** If you specified a custom cert, you will also need to specify `additionalTrustedCAs=true`
```
This will also need to be moved to a more appropriate section. Right now, the bulk of the certificate management docs are in steps 2 and 4, which can make it easy to miss this section. Without this section, the private certs will not. I'm thinking that duplicating this or something similar between this section and section 4 would be helpful.

MbolotSuse replied:
Thank you for the suggestion, change made. I've also added a small section right at the end of step 2 which indicates that you need to do this as well. That way there's something in both places.

MbolotSuse commented:
@jtravee 

> Hi @MbolotSuse, do you think we could place these docs in [our AWS Marketplace Quick Start](https://rancher.com/docs/rancher/v2.6/en/quick-start-guide/deployment/amazon-aws-marketplace-qs/) or otherwise in the installation docs instead?

I defer to your experience on which location in the docs is best for this. That being said, I don't think that this qualifies as "quick start" documentation. I think putting this under "installation" as an immediate subsection would work though. Let me know what you prefer and I'll make that change.

> In addition, please change your PR to be from staging rather than master as it will apply to an upcoming release. Thanks!

Will do.

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/adapter-uninstall/_index.md:
```diff
@@ -0,0 +1,22 @@
+weight: 3
+---
+
+### 1. Uninstall the adapter chart using helm.
```
Without the title indicators, the file is unable to render properly and lists each step as step 1. This is because there is significant spacing between each step and the command used in that step. In addition, other files (including ones in this PR) use this "title numbering scheme". Because of these factors, I am going to leave this (and the other steps in the file) as is to preserve proper order and formatting.

If you disagree please let me know, I am open to discussing further.

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/adapter-uninstall/_index.md:
````diff
@@ -0,0 +1,22 @@
+helm uninstall rancher-csp-adapter -n cattle-csp-adapter-system
+```
+
+### 2. Remove the namespace created for the adapter.
````
Will not make this change in line with https://github.com/rancher/docs/pull/4143/files#r914200355.

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/adapter-install/_index.md:
```diff
@@ -0,0 +1,95 @@
+
+### 4. Managing Certificate Updates
+
+If you had to create a secret storing a custom cert in step 2, you will need to update this secret over time as the certificate is rotated. 
```
Thank you for the suggestion, change made

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/adapter-install/_index.md:
```diff
@@ -0,0 +1,95 @@
+
+Then, follow the original installation steps to replace the content of the secret with the updated value.
+
+Last, restart the rancher-csp-adapter deployment to ensure that the updated value is made available to the adapter.
```
Thank you for the suggestion, change made

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/adapter-install/_index.md:
````diff
@@ -0,0 +1,95 @@
+  roleName: $MY_ROLE_NAME
+```
+
+> **Note:** The account number needs to be specified in a string format, exactly like the above, or the installation will fail 
````
Thank you for the suggestion, change made

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/adapter-install/_index.md:
```diff
@@ -0,0 +1,95 @@
+
+> **Note:** If you use shell variables, do not specify quotation marks. For example, MY_ACC_NUM=123456789012 will work, but MY_ACC_NUM="123456789012" will fail.
+
+Alternatively, you can use a `values.yaml` and specify options like the below, replacing `$MY_ACC_NUM` with your aws account number and `$MY_ROLE_NAME` with the name of the role created in the pre-requisites: 
```
Thank you for the suggestion, change made

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/adapter-install/_index.md:
```diff
@@ -0,0 +1,95 @@
+
+> **Note:** If you specified a custom cert, you will also need to specify `additionalTrustedCAs=true`
+
+You can do this using the helm cli using the following command, replacing `$MY_ACC_NUM` with your aws account number and `$MY_ROLE_NAME` with the name of the role created in the pre-requisites:
```
Thank you for the suggestion, change made

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/adapter-install/_index.md:
````diff
@@ -0,0 +1,95 @@
+helm repo add rancher-charts https://charts.rancher.io
+```
+
+Next, install the CSP adapter. You must specify several values, including: the account number, and the name of the role created in the pre-requisites.
````
Thank you for the suggestion, change made.

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/adapter-install/_index.md:
```diff
@@ -0,0 +1,95 @@
+
+### 3. Install the chart
+
+First, Add the `rancher/charts` repo using the following command:
```
Thank you for the suggestion, change made.

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/adapter-install/_index.md:
```diff
@@ -0,0 +1,95 @@
+
+> **Note:** Do not change the names of the file or of the created secret. Making changes to these values may result in errors when the adapter runs.
+
+### 3. Install the chart
```
Thank you for the suggestion, change made.

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/adapter-install/_index.md:
````diff
@@ -0,0 +1,95 @@
+kubectl -n cattle-csp-adapter-system create secret generic tls-ca-additional --from-file=ca-additional.pem
+```
+
+> **Note:** Do not change the names of the file or of the created secret. Making changes to these values may result in errors when the adapter runs.
````
Thank you for the suggestion, change made.

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/adapter-install/_index.md:
````diff
@@ -0,0 +1,95 @@
+kubectl get secret tls-rancher -n cattle-system -o jsonpath="{.data.tls\.crt}" | base64 -d  >> ca-additional.pem
+```
+
+Then, create a secret which uses this cert.
````
Thank you for the suggestion, change made.

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/adapter-install/_index.md:
```diff
@@ -0,0 +1,95 @@
+
+However, if your rancher install uses a custom certificate (such as a rancher-generated certificate or one signed by a private Certificate Authority), you will need to provide the certificate for this authority in pem-encoded format so that the adapter can communicate with rancher.
+
+First, retrieve the certificate that rancher is using and place in a file named `ca-additional.pem`. If you are using the rancher-generated certs option, this can be done with the following command:
```
Thank you for the suggestion, change made.

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/adapter-install/_index.md:
```diff
@@ -0,0 +1,95 @@
+
+If your rancher install uses a certificate signed by a recognized Certificate Authority (such as Let's Encrypt), then you can safely skip to the next step.
+
+However, if your rancher install uses a custom certificate (such as a rancher-generated certificate or one signed by a private Certificate Authority), you will need to provide the certificate for this authority in pem-encoded format so that the adapter can communicate with rancher.
```
Thank you for the suggestion, change made.

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/adapter-install/_index.md:
```diff
@@ -0,0 +1,95 @@
+
+The adapter requires access to the root CA that rancher is using to communicate with the rancher server. You can read more about which certificate options rancher supports in the [chart options page]({{<baseurl>}}/rancher/v2.6/en/installation/install-rancher-on-k8s/chart-options).
+
+If your rancher install uses a certificate signed by a recognized Certificate Authority (such as Let's Encrypt), then you can safely skip to the next step.
```
Thank you for the suggestion, change made.

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/adapter-install/_index.md:
```diff
@@ -0,0 +1,95 @@
+
+### 2. Create Certificate Secrets 
+
+The adapter requires access to the root CA that rancher is using to communicate with the rancher server. You can read more about which certificate options rancher supports in the [chart options page]({{<baseurl>}}/rancher/v2.6/en/installation/install-rancher-on-k8s/chart-options).
```
Thank you for the suggestion, change made.

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/adapter-install/_index.md:
```diff
@@ -0,0 +1,95 @@
+weight: 2
+---
+
+> **Note:** If you are attempting to re-install the adapter, you may experience errant out-of-compliance messages for up to an hour 
```
Thank you for the suggestion, change made.

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/_index.md:
```diff
@@ -0,0 +1,32 @@
+
+Yes. However, each cluster that rancher is installed in will need to adhere to the pre-requisites. 
+
+In addition, keep in mind that a given entitlement can only be used by one rancher management server at a time. 
```
Thank you for the suggestion, change made.

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/_index.md:
```diff
@@ -0,0 +1,32 @@
+
+**Can I use multiple instances of rancher in the same AWS account?**
+
+Yes. However, each cluster that rancher is installed in will need to adhere to the pre-requisites. 
```
Thank you for the suggestion, change made.

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/_index.md:
```diff
@@ -0,0 +1,32 @@
+
+Yes. Simply go to the AWS marketplace entry that you used to initially purchase support and increase the number of entitlements.
+
+**Can I use multiple instances of rancher in the same AWS account?**
```
Thank you for the suggestion, change made.

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/_index.md:
```diff
@@ -0,0 +1,32 @@
+
+# How to use
+1. Complete the [pre-requisite steps]({{<baseurl>}}/v2.6/en/cloud-marketplace/aws/prerequisites)
+2. [Install the CSP Adapter]({{<baseurl>}}/v2.6/en/cloud-marketplace/aws/adapter-install).
```
Thank you for the suggestion, change made.

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/_index.md:
```diff
@@ -0,0 +1,32 @@
+- You may need additional setup to support proxy/airgap use cases. See the pre-requisites for more information
+
+# How to use
+1. Complete the [pre-requisite steps]({{<baseurl>}}/v2.6/en/cloud-marketplace/aws/prerequisites)
```
Thank you for the suggestion, change made.

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/_index.md:
```diff
@@ -0,0 +1,32 @@
+- You must purchase at least one entitlement to rancher support through AWS Marketplace.
+- You may need additional setup to support proxy/airgap use cases. See the pre-requisites for more information
+
+# How to use
```
Thank you for the suggestion, change made.

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/_index.md:
```diff
@@ -0,0 +1,32 @@
+- Rancher must be deployed with additional metrics enabled
+- Rancher must be installed on an EKS cluster
+- You must purchase at least one entitlement to rancher support through AWS Marketplace.
+- You may need additional setup to support proxy/airgap use cases. See the pre-requisites for more information
```
Thank you for the suggestion, change made.

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/_index.md:
```diff
@@ -0,0 +1,32 @@
+
+- You must be running rancher v2.6.6 or higher
+- Rancher must be deployed with additional metrics enabled
+- Rancher must be installed on an EKS cluster
```
Thank you for the suggestion, change made.

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/_index.md:
```diff
@@ -0,0 +1,32 @@
+# Limitations
+
+- You must be running rancher v2.6.6 or higher
+- Rancher must be deployed with additional metrics enabled
```
Thank you for the suggestion, change made.

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/_index.md:
```diff
@@ -0,0 +1,32 @@
+
+# Limitations
+
+- You must be running rancher v2.6.6 or higher
```
Thank you for the suggestion, change made.

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/_index.md:
```diff
@@ -0,0 +1,32 @@
+
+# Overview
+
+Rancher offers an integration with the AWS marketplace which allows users to easily purchase a support contract with SUSE. This integration allows you easily adjust your support needs as you start to support more clusters.
```
Thank you for the suggestion, change made.

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/common-issues/_index.md:
```diff
@@ -0,0 +1,29 @@
+
+This error indicates that while the adapter was installed into the cluster, an error has occurred which prevents it from properly checking-in/checking-out licenses.
+
+Oftentimes, this is because the IAM role was not setup properly. Review the [pre-requisites]({{<baseurl>}}/rancher/v2.6/en/cloud-marketplace/aws/prerequisites) and verify that:
```
Thank you for the suggestion, change made

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/common-issues/_index.md:
```diff
@@ -0,0 +1,29 @@
+
+Oftentimes, this is because the IAM role was not setup properly. Review the [pre-requisites]({{<baseurl>}}/rancher/v2.6/en/cloud-marketplace/aws/prerequisites) and verify that:
+
+- An OIDC provider has been created/associated with the cluster rancher is running on
```
Thank you for the suggestion, change made

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/common-issues/_index.md:
```diff
@@ -0,0 +1,29 @@
+Oftentimes, this is because the IAM role was not setup properly. Review the [pre-requisites]({{<baseurl>}}/rancher/v2.6/en/cloud-marketplace/aws/prerequisites) and verify that:
+
+- An OIDC provider has been created/associated with the cluster rancher is running on
+- The IAM role has been configured to trust this OIDC provider
```
Thank you for the suggestion, change made

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/common-issues/_index.md:
```diff
@@ -0,0 +1,29 @@
+
+- An OIDC provider has been created/associated with the cluster rancher is running on
+- The IAM role has been configured to trust this OIDC provider
+- The IAM role has at least the permissions outlined in the policy
```
Thank you for the suggestion, change made

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/common-issues/_index.md:
```diff
@@ -0,0 +1,29 @@
+- The IAM role has been configured to trust this OIDC provider
+- The IAM role has at least the permissions outlined in the policy
+
+If these have been configured correctly, reach out to support.
```
Thank you for the suggestion, change made

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/common-issues/_index.md:
```diff
@@ -0,0 +1,29 @@
+
+If these have been configured correctly, reach out to support.
+
+**I see a banner message which states "AWS Marketplace Adapter: You have exceeded your licensed node count. At least x more license(s) are required in AWS to become compliant"**
```
Thank you for the suggestion, change made

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/common-issues/_index.md:
```diff
@@ -0,0 +1,29 @@
+
+**I see a banner message which states "AWS Marketplace Adapter: You have exceeded your licensed node count. At least x more license(s) are required in AWS to become compliant"**
+
+This message indicates that you do not have enough entitlements for the amount of nodes rancher is currently managing.
```
Thank you for the suggestion, change made

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/common-issues/_index.md:
```diff
@@ -0,0 +1,29 @@
+
+Keep in mind the following limitations:
+
+- Each entitlement is valid for certain number of nodes
```
Thank you for the suggestion, change made

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/common-issues/_index.md:
```diff
@@ -0,0 +1,29 @@
+Keep in mind the following limitations:
+
+- Each entitlement is valid for certain number of nodes
+- Every node currently managed by rancher counts toward your usage total, even nodes in the cluster rancher is installed on
```
Thank you for the suggestion, change made

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/common-issues/_index.md:
```diff
@@ -0,0 +1,29 @@
+
+- Each entitlement is valid for certain number of nodes
+- Every node currently managed by rancher counts toward your usage total, even nodes in the cluster rancher is installed on
+- Each entitlement can be used by at most 1 rancher instance. For example, if you have two running rancher instances in your account (each installed on a separate EKS cluster), then you will need at least two entitlements.
```
Thank you for the suggestion, change made

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/common-issues/_index.md:
```diff
@@ -0,0 +1,29 @@
+- Every node currently managed by rancher counts toward your usage total, even nodes in the cluster rancher is installed on
+- Each entitlement can be used by at most 1 rancher instance. For example, if you have two running rancher instances in your account (each installed on a separate EKS cluster), then you will need at least two entitlements.
+
+You may also have recently uninstalled/re-installed the adapter. If the adapter looses track of the licenses that it is currently managing, it can take up to an hour to resolve the actual state of the licenses.
```
Thank you for the suggestion, change made

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/prerequisites/_index.md:
```diff
@@ -0,0 +1,97 @@
+---
+title: Pre-Requisites
```
Thank you for the suggestion, change made

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/prerequisites/_index.md:
```diff
@@ -0,0 +1,97 @@
+
+### 1. Setting Up License Manager and Purchasing Support
+
+First, Complete the [first step](https://docs.aws.amazon.com/license-manager/latest/userguide/getting-started.html) of the license manager one-time setup.
```
Thank you for the suggestion, change made

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/prerequisites/_index.md:
```diff
@@ -0,0 +1,97 @@
+### 1. Setting Up License Manager and Purchasing Support
+
+First, Complete the [first step](https://docs.aws.amazon.com/license-manager/latest/userguide/getting-started.html) of the license manager one-time setup.
+Next, go to the AWS marketplace. Locate the "Rancher Premium Support Billing Container Starter Pack". Purchase at least one entitlement.
```
Thank you for the suggestion, change made

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/prerequisites/_index.md:
```diff
@@ -0,0 +1,97 @@
+First, Complete the [first step](https://docs.aws.amazon.com/license-manager/latest/userguide/getting-started.html) of the license manager one-time setup.
+Next, go to the AWS marketplace. Locate the "Rancher Premium Support Billing Container Starter Pack". Purchase at least one entitlement.
+
+If you have installed rancher using the "Rancher Setup" AWS Marketplace offering, skip to step 4.
```
Thank you for the suggestion, change made

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/prerequisites/_index.md:
```diff
@@ -0,0 +1,97 @@
+Follow the [rancher docs]({{<baseurl>}}/rancher/v2.6/en/installation/install-rancher-on-k8s/amazon-eks/) to create an EKS cluster. When you get to the final step to install rancher, stop and return to this page. This cluster will need to meet the following requirements:
+
+- EKS version 1.22
+- Each node in the cluster has access to the registry containing rancher and its related images
```
Thank you for the suggestion, change made

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/prerequisites/_index.md:
```diff
@@ -0,0 +1,97 @@
+- EKS version 1.22
+- Each node in the cluster has access to the registry containing rancher and its related images
+- Each node in the cluster has access to the ECR repo storing the CSP Adapter
+- Each node in the cluster has access to the license manager service
```
Thank you for the suggestion, change made

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/prerequisites/_index.md:
```diff
@@ -0,0 +1,97 @@
+- Each node in the cluster has access to the registry containing rancher and its related images
+- Each node in the cluster has access to the ECR repo storing the CSP Adapter
+- Each node in the cluster has access to the license manager service
+- Each node in the cluster has access to global endpoints for the STS service
```
Thank you for the suggestion, change made

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/prerequisites/_index.md:
```diff
@@ -0,0 +1,97 @@
+### 3. Install Rancher
+
+In addition to the options specified to install rancher in step 8 of the [rancher docs]({{<baseurl>}}/rancher/v2.6/en/installation/install-rancher-on-k8s/amazon-eks/), you will also need to enable extra metrics. 
+This can be done through the helm cli through the following options: 
```
Thank you for the suggestion, changes made

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/aws/prerequisites/_index.md:
```diff
@@ -0,0 +1,97 @@
+
+An IAM role is required for the CSP adapter to check-in/check-out entitlements. 
+
+First, Configure the trust policy as below, replacing `MY_AWS_ACC` with your aws account number, `MY_AWS_REGION` with your aws region, and `MY_OIDC_PROVIDER` with the id of your OIDC provider:
```
Thank you for the suggestion, change made

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/supportconfig/_index.md:
```diff
@@ -0,0 +1,54 @@
+
+After installing the CSP adapter, you will have the ability to generate a supportconfig bundle. This bundle is a tar file which can be used to quickly provide information to support.
+
+These bundles can be created through rancher or through direct access to the cluster rancher is installed on (accessing through rancher is preferred).
```
Thank you for the suggestion, change made

MbolotSuse commented on content/rancher/v2.6/en/cloud-marketplace/supportconfig/_index.md:
```diff
@@ -0,0 +1,54 @@
+
+These bundles can be created through rancher or through direct access to the cluster rancher is installed on (accessing through rancher is preferred).
+
+> **Note:** Only admin users can generate/download supportconfig bundles, regardless of method 
```
Thank you for the suggestion, change made

MbolotSuse commented:
@btat @jtravee I've made the following changes:
- Rebased off of staging
- Changed PR target to staging
- Made changes requested in review

This is now ready for review again. Please let me know if you think anything else should be changed.

jtravee commented:
Please add a note to 2.6.7 RN for new features.

MbolotSuse commented:
@jtravee @btat Is this still clear to merge? Sorry for the delay

jtravee commented:
> @jtravee @btat Is this still clear to merge? Sorry for the delay

Yes, no problem!
